moto/ec2/models.py

```diff
@@ -6128,6 +6143,98 @@ def search_transit_gateway_routes(self, transit_gateway_route_table_id, filters,
         return routes
 
 
+class TransitGatewayAttachment(TaggedEC2Resource):
+
+    def __init__(
+        self,
+        backend,
+        resource_id,
+        resource_type,
+        transit_gateway_id,
+        tags=[]
+    ):
+
+        self.ec2_backend = backend
+        self.association = {}
+        self.resource_id = resource_id
+        self.resource_type = resource_type
+
+        self.id = random_transit_gateway_attachment_id()
+        self.transit_gateway_id = transit_gateway_id
+
+        self.state = "available"
+        self.add_tags(tags or {})
+
+        self._created_at = datetime.utcnow()
+
+    @property
+    def create_time(self):
+        return iso_8601_datetime_with_milliseconds(self._created_at)
+
+    @property
+    def resource_owner_id(self):
+        return ACCOUNT_ID
+
+    @property
+    def transit_gateway_owner_id(self):
+        return ACCOUNT_ID
+
+
+class TransitGatewayVpcAttachment(TransitGatewayAttachment):
+
+    DEFAULT_OPTIONS = {
+        "ApplianceModeSupport": "disable",
+        "DnsSupport": "enable",
+        "Ipv6Support": "disable"
+    }
+
+    def __init__(
+        self,
+        backend,
+        transit_gateway_id,
+        vpc_id,
+        subnet_ids,
+        tags=[],
+        options={}
+    ):
+
+        super().__init__(
+            backend=backend,
+            transit_gateway_id=transit_gateway_id,
+            resource_id=vpc_id,
+            resource_type="vpc",
+            tags=tags,
+        )
+
+        self.vpc_id = vpc_id
+        self.subnet_ids = subnet_ids
+        self.options = patch_dict(self.DEFAULT_OPTIONS, options)
```

Comment:
Same as above, regarding swapping the order of arguments.

Reply:
Created new method